patches/chrome-browser-resources-settings-clear_browsing_data_dialog-clear_browsing_data_dialog.js.patch

```diff
@@ -0,0 +1,80 @@
+diff --git a/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.js b/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.js
+index ba8db3241392d66e429a66dba47a5b740f3dfd28..ef8afbeace3a6c835426b3a9ccc97e937a89702b 100644
+--- a/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.js
++++ b/chrome/browser/resources/settings/clear_browsing_data_dialog/clear_browsing_data_dialog.js
```

Comment:
we also need to do something to reduce or eliminate this patch. Checking with @yrliou because I forget what we talked about in a previous PR

Reply:
In settings.pageVisibility case, we were able to overwrite the chromium one and put all of our own logic in a separate js in https://github.com/brave/brave-core/blob/master/browser/resources/settings/brave_page_visibility.js.
So in that case, we only need to patch one line to include our js in https://github.com/brave/brave-core/blob/2dbbfc935c48d01cb409fd07d4d7b812c6642c59/patches/chrome-browser-resources-settings-page_visibility.html.patch.
I think we should be able to pull our stuff out in a separate js here too, maybe a few extra work than the page visibility case because you have some dep of clear_browsing_data_dialog/clear_browsing_data_dialog.js, but I think we could do it in a lot less patching here.

Reply:
@yrliou, I've externalized some of it. Do you have any suggestions on externalizing the remaining code?

Reply:
cc @yrliou @petemill ^^

Reply:
Possibly the best option here is to use the Polymer v1 `behaviors` property, which essentially defines a superclass that has access to all the lifecycle methods
> For lifecycle events, the lifecycle callback is called for each behavior in the order given in the behaviors array, followed by the callback on the prototype.
Source: https://polymer-library.polymer-project.org/1.0/docs/devguide/behaviors

In this case:
- L17 of the patch could be done by hooking in to `ready` and adding a handler for `this.addWebUIListener('update-counter-text'`
- L9 of the patch could be done by hooking in to the `settings-boolean-control-change` listener. I think the behavior can define its own listeners property without overriding the sub-class property, but if not then we could call `this.listen(...` in `attached`: https://polymer-library.polymer-project.org/1.0/docs/devguide/events#imperative-listeners
- L25 onwards can just be defined simply in the behavior.

Reply:
Thanks @petemill. Rolled everything into a behavior.